### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by rawmind0**
This PR contains:
* Fix `rancher2_cluster_sync` resource error if referred cluster deleted out of band. Issue #401 
* Fix `rancher2_namespace` and `rancher2_project` resources error if destroyed by a cluster member. Issue #390 and #388
* Fix `rancher2_app` resource error if referred project deleted out of band. Issue #366 
* Fix `rancher2_app` doc typo on `target_namespace` argument description. Issue #366

### CODE REVIEW

**Review by paraglade — APPROVED** [review]
approved

**Comment by armsnyder** [comment]
Thank you!

File: rancher2/resource_rancher2_cluster_sync.go [hunk 1]
```diff
@@ -75,7 +76,7 @@ func resourceRancher2ClusterSyncCreate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
-	d.SetId(clusterID)
+	d.SetId(clusterSyncPrefixID + clusterID)
```
Comment (armsnyder) [inline rancher2/resource_rancher2_cluster_sync.go:79]: @rawmind0 FYI this is a breaking change. I tried using the latest (master) build of this provider, and I had a dependency on the rancher2_cluster_sync.id, and it's broke now. I think we need to revert this before doing a release.

https://www.terraform.io/docs/extend/best-practices/versioning.html#example-major-number-increments

> Increasing the MAJOR number is intended to signify potentially breaking changes.
> Within Terraform provider development, some examples include:
> * Changing resource ID format
>

Comment (armsnyder) [reply rancher2/resource_rancher2_cluster_sync.go:79]: Moved this to an issue https://github.com/rancher/terraform-provider-rancher2/issues/419

**Pull request merged**